tests/test-autoupdater.c

```diff
@@ -0,0 +1,242 @@
+  g_autoptr(GFile) server_root = NULL;
+  g_autoptr(EosTestServer) server = NULL;
+  g_autoptr(EosTestSubserver) subserver = NULL;
+  g_autoptr(GFile) client_root = NULL;
+  g_autoptr(EosTestClient) client = NULL;
+  g_autoptr(GHashTable) additional_metadata_for_commit = NULL;
+
+  /* Arbitrarily say that the currently booted commit is version 1.0.0. */
+  eos_test_add_metadata_for_commit (&additional_metadata_for_commit,
+                                    0, "version", "1.0.0");
+
+  server_root = g_file_get_child (fixture->tmpdir, "main");
+  server = eos_test_server_new_quick (server_root,
+                                      default_vendor,
+                                      default_product,
+                                      default_collection_ref,
+                                      0,
+                                      fixture->gpg_home,
+                                      keyid,
+                                      default_ostree_path,
+                                      NULL, NULL,
+                                      additional_metadata_for_commit,
+                                      &error);
+  g_assert_no_error (error);
+  g_assert_nonnull (server);
+
+  g_assert_cmpuint (server->subservers->len, ==, 1u);
+
+  subserver = g_object_ref (EOS_TEST_SUBSERVER (g_ptr_array_index (server->subservers, 0)));
+  client_root = g_file_get_child (fixture->tmpdir, "client");
+  client = eos_test_client_new (client_root,
+                                default_remote_name,
+                                subserver,
+                                default_collection_ref,
+                                default_vendor,
+                                default_product,
+                                default_auto_bootloader,
+                                &error);
+  g_assert_no_error (error);
+  g_assert_nonnull (client);
+
+  *out_server = g_steal_pointer (&server);
+  *out_subserver = g_steal_pointer (&subserver);
+  *out_client = g_steal_pointer (&client);
+}
+
+static void
+get_poll_results (GFile    *autoupdater_dir,
+                  guint64  *last_changed_usecs,
+                  gchar   **update_refspec,
+                  gchar   **update_id)
+{
+  g_autoptr(GFile) state_dir = g_file_get_child (autoupdater_dir, "state");
+  g_autoptr(GFile) results_file = g_file_get_child (state_dir, "autoupdater-poll-results");
+  g_autoptr(GBytes) results_bytes = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GVariant) results_variant = NULL;
+
+  results_bytes = g_file_load_bytes (results_file, NULL, NULL, &error);
+  g_assert_no_error (error);
+  g_assert_nonnull (results_bytes);
+
+  results_variant = g_variant_new_from_bytes (G_VARIANT_TYPE ("a{sv}"),
+                                              results_bytes,
+                                              FALSE);
+  if (last_changed_usecs)
+    if (!g_variant_lookup (results_variant, "LastChangedUsecs", "t", last_changed_usecs))
```

Comment:
Nitpick: these `if`s could be combined:
```suggestion
  if (last_changed_usecs &&
      !g_variant_lookup (results_variant, "LastChangedUsecs", "t", last_changed_usecs))
```

Reply:
:+1: